graal/backends/core/analyzers/cloc.py

```diff
@@ -32,45 +32,99 @@ class Cloc(Analyzer):
     This class allows to call Cloc over a file, parses
     the result of the analysis and returns it as a dict.
     """
-    version = '0.1.1'
+    version = '0.2.0'
 
-    def analyze(self, **kwargs):
+    def __analyze_file(self, message):
         """Add information about LOC, blank and commented lines using CLOC
 
-        :param file_path: file path
+        :param message:
 
-        :returns result: dict of the results of the analysis
+        :returns result: dict of the results of the analysis over a file
         """
-        result = {'blanks': 0,
-                  'comments': 0,
-                  'loc': 0
-                  }
-        file_path = kwargs['file_path']
-        flag = False
 
-        try:
-            msg = subprocess.check_output(['cloc', file_path]).decode("utf-8")
-        except subprocess.CalledProcessError as e:
-            raise GraalError(cause="Cloc failed at %s, %s" % (file_path, e.output.decode("utf-8")))
-        finally:
-            subprocess._cleanup()
+        flag = False
+        results = {
+            "blanks": 0,
+            "comments": 0,
+            "loc": 0
+        }
 
-        for line in msg.split("\n"):
+        for line in message.strip().split("\n"):
             if flag:
                 if not line.startswith("-----"):
                     digested = " ".join(line.split())
                     info_file = digested.split(" ")
                     blank_lines = int(info_file[2])
                     commented_lines = int(info_file[3])
                     loc = int(info_file[4])
+                    results["blanks"] = blank_lines
+                    results["comments"] = commented_lines
+                    results["loc"] = loc
+                    break
 
-                    result['blanks'] = blank_lines
-                    result['comments'] = commented_lines
-                    result['loc'] = loc
+            if line.lower().startswith("language"):
+                flag = True
+
+        return results
+
+    def __analyze_repository(self, message):
+        """Add information LOC, Total files, blank and commented lines using CLOC
```

Comment:
Total should be `total`.
Could you complete the description with something like:
`Add information LOC, total files, blank and commented lines using CLOC for the full repository` ?